Tests/BasicTests/LockTests.swift

```diff
@@ -39,12 +39,34 @@ class LockTests: XCTestCase {
         let N = 10
         let threads = (1...N).map { idx in
             return Thread {
-                _ = try! SwiftPMProduct.TestSupportExecutable.execute(["fileLockTest", tempDir.path.pathString, sharedResource.path.pathString, String(idx)])
+                _ = try! SwiftPMProduct.TestSupportExecutable.execute(["fileLockTest", "test.lock", tempDir.path.pathString, sharedResource.path.pathString, String(idx)])
             }
         }
         threads.forEach { $0.start() }
         threads.forEach { $0.join() }
 
         XCTAssertEqual(try localFileSystem.readFileContents(sharedResource.path).description, String((N * (N + 1) / 2 )))
     }
+
+    func testFileLockTry() throws {
+        let tempDir = try TemporaryDirectory(removeTreeOnDeinit: true)
+        let fileLock = FileLock(name: "test.lock", in: tempDir.path)
+
+        let lockThread = Thread {
+            try! SwiftPMProduct.TestSupportExecutable.execute(["fileLockLock", "test.lock", tempDir.path.pathString, "3"])
```

Comment:
Given the trivial implementation, I don't think it's worth adding a test case that will sleep for 3 seconds. It seems fine to land this PR without a test for the new `try` method.